sdk/src/main/java/io/customer/sdk/CustomerIO.kt

```diff
@@ -94,23 +130,40 @@ class CustomerIO internal constructor(
                 backgroundQueueMinNumberOfTasks = 10
             )
 
-            val customerIoComponent =
-                CustomerIOComponent(customerIOConfig = config, context = appContext)
-
-            val client = CustomerIO(
-                config = config,
-                store = customerIoComponent.buildStore(),
-                api = customerIoComponent.buildApi(),
-            )
+            val client = CustomerIO()
+            client.diGraph = CustomerIOComponent().apply {
+                sdkConfig = config
+                context = appContext
+            }
 
-            activityLifecycleCallback = CustomerIOActivityLifecycleCallbacks(client)
+            activityLifecycleCallback = CustomerIOActivityLifecycleCallbacks(client, config)
             appContext.registerActivityLifecycleCallbacks(activityLifecycleCallback)
 
             instance = client
+
             return client
         }
     }
 
+    // Since this class is at the top-most level of the Tracking module,
+    // we get instances from the DiGraph, not through constructor dependency injection.
+    lateinit var diGraph: CustomerIOComponent
```

Comment:
I originally made this change to make the `CustomerIO` constructor empty when I did the work to allow customers to create instances of `CustomerIO`. Now that I removed that, I am OK with going back with the original implementation. We can use the constructor of `CustomerIO` since it's `internal`. 

I am fine with either way. I can move it back.

Reply:
Complete.